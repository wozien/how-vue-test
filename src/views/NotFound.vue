<template>
  <div class="container">
    <h1>404</h1>
    <p>The page you're looking for could not be found. Try selecting one of the lists in the nav bar.</p>
  </div>
</template>

<script>
export default {
  HTTPStatus: 404
}
</script>

<style>
  .container {
    background-color: #fff;
    border-radius: 2px;
    position: absolute;
    margin: 70px 0;
    width: 100%;
    text-align: center;
    padding: 50px 0;
  }
</style>