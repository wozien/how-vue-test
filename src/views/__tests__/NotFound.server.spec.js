/**
 * @jest-environment node
 */
import { renderToString, render } from '@vue/server-test-utils'
import NotFound from '../NotFound.vue'

describe('NotFound.vue', () => {
  test('renders correctly on server', async () => {
    const html = await renderToString(NotFound)
    expect(html).toMatchSnapshot()
  })

  test('renders 404 inside <h1>', async () => {
    const wrapper = await render(NotFound)
    expect(wrapper.find('h1').text()).toBe('404')
  })
})