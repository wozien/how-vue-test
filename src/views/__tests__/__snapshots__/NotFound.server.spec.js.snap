// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound.vue renders correctly on server 1`] = `"<div data-server-rendered=\\"true\\" class=\\"container\\"><h1>404</h1> <p>The page you're looking for could not be found. Try selecting one of the lists in the nav bar.</p></div>"`;
